ciao-controller/internal/datastore/datastore.go

```diff
@@ -1406,6 +1406,29 @@ func (ds *Datastore) AddBlockDevice(device types.BlockData) error {
 	return nil
 }
 
+func (ds *Datastore) DeleteBlockDevice(ID string) error {
+	// lock both tenants and devices maps
+	var err error
+
+	ds.bdLock.Lock()
+	ds.tenantsLock.Lock()
+
+	dev, ok := ds.blockDevices[ID]
+	if ok {
+		delete(ds.blockDevices, ID)
+		delete(ds.tenants[dev.TenantID].devices, ID)
+
+		go ds.db.deleteBlockData(ID)
```

Comment:
I'll move the go call as you suggested to outside the loop. that was actually a mistake anyway and an artifact of when I had this sqlite update happening synchronously.